graph/kosaraju.go

```diff
@@ -0,0 +1,132 @@
+package graph
+
+import (
+	"github.com/abaron10/Gothon/gothonSlice"
+)
+
+type graph struct {
+	nodes    []int
+	reversed *graph
+	edges    map[int][]*edge
+}
+
+type edge struct {
+	from int
+	to   int
+}
+
+func newEdge(from int, to int) *edge {
+	return &edge{from: from, to: to}
+}
+
+func NewGraphK() *graph {
+	reversedGraph := &graph{nodes: []int{}, edges: map[int][]*edge{}}
+	return &graph{nodes: []int{}, edges: map[int][]*edge{}, reversed: reversedGraph}
+}
+
+func (g *graph) AddEdge(from int, to int) {
+	edges, _ := g.edges[from]
+	edges = append(edges, newEdge(from, to))
+
+	g.edges[from] = edges
+	g.nodes = append(g.nodes, from)
+
+	//Computing transposed graph at the same time user adds an Edge.
+	if g.reversed != nil {
+		g.reversed.AddEdge(to, from)
+	}
+}
+
+func (g *graph) EvaluateSCC() [][]int {
+	baseOrder := g.calculateStackBaseOrder()
+	sccResult := g.findSCCComponents(baseOrder)
+	return g.computeAnswer(sccResult)
+}
+
+func (g *graph) calculateStackBaseOrder() []int {
+	visited := map[int]struct{}{}
+	stack := []int{}
+
+	for _, node := range g.nodes {
+		if _, ok := visited[node]; !ok {
+			g.dfs(node, visited, &stack)
+		}
+	}
+
+	return stack
+}
+
+func (g *graph) findSCCComponents(orderNodes []int) map[int]int {
+	visited := map[int]struct{}{}
+	scc := map[int]int{}
+	counter := 0
+
+	for len(orderNodes) > 0 {
+		//using https://github.com/abaron10/Gothon library to emulate stack popping as Python do
+		node := gothonSlice.Pop(&orderNodes, -1)
+		if _, ok := visited[node]; !ok {
+			g.dfsReversed(node, visited, scc, counter)
+			counter++
+		}
+	}
+
+	return scc
+}
+
+/*
+---------------------------------------------------------------------------------------------------------
+--------------------------------DFS IMPLEMENTATIONS (BASE AND TRANSPOSED GRAPH)---------------------------
+-----------------------------------------------------------------------------------------------------------
+*/
```

Comment:
The styling of these comments don't follow Go standards, please refrain from useless commenting since it doesn't provide any real value to the reader.